### PR TITLE
Xeno get a message when silos/turrets are built. It is also registered in the game log

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Title

## Why It's Good For The Game

So you can know when a hivelord is puttin turrets everywhere like it's TF2. 

## Changelog
:cl:
qol: Xeno message is sent to the hive when silos/turrets are built
admin: Psy points spent are now saved in game log
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 1]
```diff
@@ -1106,6 +1109,8 @@
 	new /obj/structure/resin/xeno_turret(get_turf(A), X.hivenumber)
 
 	SSpoints.xeno_points_by_hive[X.hivenumber] -= psych_cost
+	log_game("[owner] built a turret in get_area(A), spending [psych_cost] psy points in the process")
+	xeno_message("[X.name] has built a new turret get_area(A)!", "xenoannounce", 5, X.hivenumber)
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:1113]:
```suggestion
	xeno_message("[X.name] has built a new turret at [get_area(A)]!", "xenoannounce", 5, X.hivenumber)
```

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 2]
```diff
@@ -1031,11 +1031,14 @@
 	
 	to_chat(owner, "<span class='notice'>We build a new silo for [final_psych_cost] psy points.</span>")
 	SSpoints.xeno_points_by_hive[X.hivenumber] -= final_psych_cost
+	log_game("[owner] has built a silo in [get_area(A)], spending [final_psych_cost] psy points in the process")
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:1034]: areacoord

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 3]
```diff
@@ -1106,6 +1109,8 @@
 	new /obj/structure/resin/xeno_turret(get_turf(A), X.hivenumber)
 
 	SSpoints.xeno_points_by_hive[X.hivenumber] -= psych_cost
+	log_game("[owner] built a turret in get_area(A), spending [psych_cost] psy points in the process")
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:1112]: A) this doesnt work you forgot the []
B) areacoord

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Review by TiviPlus — APPROVED** [review]

**Comment by SilencedMP5A5** [comment]
next we'll get an alert whenever someone breaks a resin door :^)

**Review by TiviPlus — APPROVED** [review]

**Comment by bowlofcereal** [comment]
this is good so you can see which motherfucker is placing 3000 turrets and preventing you from building a pool the entire game

**Pull request merged**